model/src/gateway/event/dispatch.rs

```diff
@@ -101,6 +102,59 @@ impl DispatchEvent {
     }
 }
 
+impl TryFrom<Event> for DispatchEvent {
+    type Error = EventConversionError<Event>;
+
+    fn try_from(event: Event) -> Result<Self, Self::Error> {
+        match event {
```

Comment:
This would look a lot cleaner by wrapping the match in a single Ok and doing a `return Err` below